Src/ILGPU/IR/Transformations/Optimizer.cs

```diff
@@ -164,16 +160,18 @@ public static class Optimizer
             // Lower all value structures that could have been created during the
             // following passes:
             // LowerPointerViews, AcceleratorSpecializer and AddressSpaceSpecializer
-            builder.Add(new LowerStructures());
+            if (level > OptimizationLevel.O0)
+                builder.Add(new LowerStructures());
 
-            // Apply DCE phase in release mode to remove all dead values that
-            // could be created in prior passes
+            // Apply DCE phase to remove all dead values that could be created in prior
+            // passes
+            builder.Add(new UnreachableCodeElimination());
```

Comment:
As we discussed, I fully agree that we need to apply these transformations in all cases. However, I still believe that the additional "DCE" pass in debug mode is not required. Am I right about that?